intera_interface/src/intera_motion_interface/motion_controller_action_client.py

```diff
@@ -54,12 +54,17 @@ def send_trajectory(self, trajectory):
         goal.trajectory = trajectory
         self._client.send_goal(goal)
 
-    def wait_for_result(self):
+    def wait_for_result(self, timeout=None):
         """
         Wait for the current task to finish and then return the result
+        @param timeout: maximum time to block before returning
         @return: the result of the MotionCommand.action
         """
-        self._client.wait_for_result()
+        if timeout is None:
+            rosTimeout = rospy.Duration()
+        else:
+            rosTimeout = rospy.Duration(timeout)
+        self._client.wait_for_result(rosTimeout)
```

Comment:
This case structure is a bit complicated and overrides the existing  default behavior inside of `wait_for_result()`. I suggest something like:
```
if timeout is None:
    self._client.wait_for_result()
else:
    self._client.wait_for_result(rospy.Duration(timeout))
```